Carebase-Solutions/carebase-android
# [BUGFIX] Fixed procedure equipment entry and refactored code for one procedure only entry

### What is the current behaviour?

- When creating a procedure and adding equipment already in the inventory, app crashes because of ill formed data being passes between fragments. #31 
- If an equipment was used in more than one procedure, users can enter another procedure and identify the equipment used in both procedures.

### What is the new behaviour?
- Procedure entry now works whether or not equipment is already in the inventory. 
- Also removed add procedure button and multi procedure entry as it did not work and may not be very intuitive.

### comments
To fix this issue I refactored the code, for more readability keeping in mind that a greater overhaul is need in the procedure feature.

### How to test
1. Create a procedure
2. Enter procedure details
3. Add equipment not in inventory
4 (and) Add equipment in inventory
5. Check that procedure was successfully created in the equipment's usage section
